app/helpers/repository_helper.rb

```diff
@@ -105,7 +105,11 @@ def remove_files(files, user, repo, path: '/', txn: nil, keep_folder: true)
     end
 
     current_path = Pathname.new path
-
+    files.each do |file_name|
+      relative_path = Pathname.new(file_name)
+      abs_path = File.join(current_path, relative_path)
+      return [true, [:file_not_exist]] unless repo.get_latest_revision.path_exists?(abs_path)
```

Comment:
Overall, you should merge this logic into the loop below, which already loops through each file and constructs a `file_path` (which you should be able to pass into `repo.get_latest_revision.path_exists?`).

Also, it appears from the documentation of this method that the first return value (the boolean) means different things depending on whether `txn` is passed in or not. It's counter-intuitive, but it seems to indicate success/failure in opposite ways depending on whether `txn` is passed in. You should maintain the same behaviour for the new `return` statement as well.

Then, you should change the `success` variable name in the `submission_files_controller.rb` (I don't think it actually means "success")